[Refactor/Chore] missing-attribute in CI

### Self Checks

- [x] I have read the [Contributing Guide](https://github.com/langgenius/dify/blob/main/CONTRIBUTING.md) and [Language Policy](https://github.com/langgenius/dify/issues/1542).
- [x] This is only for refactors or chores; if you would like to ask a question, please head to [Discussions](https://github.com/langgenius/dify/discussions/categories/general).
- [x] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [x] I confirm that I am using English to submit this report, otherwise it will be closed.
- [x] 【中文用户 & Non English User】请使用英语提交，否则会被关闭 ：）
- [x] Please do not modify this template :) and fill in all the required fields.

### Description

a main CI error:
```
Run make type-check
  make type-check
  shell: /usr/bin/bash -e {0}
  env:
    UV_PYTHON_INSTALL_DIR: /home/runner/work/_temp/uv-python-dir
    UV_PYTHON: 3.12
📝 Running type checks (basedpyright + pyrefly + mypy)...
+++ realpath ./dev/basedpyright-check
++ dirname /home/runner/work/dify/dify/dev/basedpyright-check
+ SCRIPT_DIR=/home/runner/work/dify/dify/dev
+ cd /home/runner/work/dify/dify/dev/..
+ PATH_TO_CHECK=
+++ uname -s
++ [[ Linux == \D\a\r\w\i\n ]]
++ nproc
+ CPU_CORES=4
+ uv run --directory api --dev -- basedpyright --threads 4
0 errors, 0 warnings, 0 notes
ERROR Object of class `object` has no attribute `get` [missing-attribute]
  --> events/event_handlers/update_app_dataset_join_when_app_model_config_updated.py:59:29
make: *** [Makefile:73: type-check] Error 1
Error: Process completed with exit code 2.
```

https://github.com/langgenius/dify/actions/runs/22745537836/job/65968418828

### Motivation

_No response_

### Additional Context

_No response_

--- a/api/events/event_handlers/update_app_dataset_join_when_app_model_config_updated.py
+++ b/api/events/event_handlers/update_app_dataset_join_when_app_model_config_updated.py
@@ -1,3 +1,5 @@
+from typing import Any, cast
+
 from sqlalchemy import select
 
 from events.app_event import app_model_config_was_updated
@@ -54,9 +56,11 @@ def get_dataset_ids_from_model_config(app_model_config: AppModelConfig) -> set[s
             continue
 
         tool_type = list(tool.keys())[0]
-        tool_config = list(tool.values())[0]
+        tool_config = cast(dict[str, Any], list(tool.values())[0])
         if tool_type == "dataset":
-            dataset_ids.add(tool_config.get("id"))
+            dataset_id = tool_config.get("id")
+            if isinstance(dataset_id, str):
+                dataset_ids.add(dataset_id)
 
     # get dataset from dataset_configs
     dataset_configs = app_model_config.dataset_configs_dict

